storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -395,7 +418,11 @@ private RemoteIndexCache.Entry createCacheEntry(RemoteLogSegmentMetadata remoteL
                 }
             });
 
-            return new Entry(offsetIndex, timeIndex, txnIndex);
+            Entry entry = new Entry(offsetIndex, timeIndex, txnIndex);
+            int entrySize = estimatedEntrySize(entry);
+            entry.setEntrySize(entrySize);
```

Comment:
instead of exposing a public function to set the entry size, perhaps it will be better to keep it private and calculate in constructor. Asking because there is no use case where an entity outside `Entry` will ever possibly want to set entry size based on it's own calculation.